### PR TITLE
Add athene2 minikube

### PR DESCRIPTION
**Pull request by rbock44**
- refactor Makefiles to fit p_data Makefile style
- add liveness probe for athene2 app
- fix broken editor and legacy editor in definitions.php.tpl file
- pull out resources for legacy editor and editor to reduce resources in minikube
- introduce image_pull_policy to support Never for minikube
- add two helper container docker images dbsetup and dbdump which are useful for kpi and athene2
- remove cloudsql proxy parts in kpi module
- push dbsetup and dbdump to be own module
- replace httpd.conf with override-httpd.conf which only has deltas currently empty

### CODE REVIEW

File: container/dbdump/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,14 @@
+FROM alpine
+RUN apk update && apk add mysql-client && rm -rf /var/cache/apk/*
+RUN mkdir /etc/periodic/midnight
+ADD ./scripts/cron.sh /tmp/
```
Comment (inyono) [inline container/dbdump/Dockerfile]: Why can't / don't we use Kubernetes cronjobs?

File: live/dev/main.tf [hunk 2]
```diff
@@ -149,6 +151,7 @@ module "athene2" {
   php_container_limits_memory   = "400Mi"
   php_container_requests_cpu    = "250m"
   php_container_requests_memory = "200Mi"
+  liveness_probe_host           = "de.${local.domain}"
```
Comment (inyono) [inline live/dev/main.tf]: Health check works w/o subdomain even, so `local.domain` should also work

File: minikube/athene2/main.tf [hunk 3]
```diff
@@ -0,0 +1,133 @@
+#####################################################################
+# settings for dev
+#####################################################################
+locals {
+  environment      = "dev"
+  domain_serlo     = "serlo.local"
+  domain_env_serlo = "${local.environment}.${local.domain_serlo}"
```
Comment (inyono) [inline minikube/athene2/main.tf]: Not true anymore, is it? All our systems now run on root TLDs? Or what's the different between `domain_env_serlo` and `domain_serlo`?

File: minikube/athene2/main.tf [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+
+  container_limits_cpu      = "200m"
+  container_limits_memory   = "200Mi"
+  container_requests_cpu    = "100m"
+  container_requests_memory = "100Mi"
+}
+
+module "editor-renderer" {
+  source            = "./../../modules/editor-renderer"
+  image             = "serlo/athene2-editor-renderer:latest"
```
Comment (inyono) [inline minikube/athene2/main.tf]: The docker image is called `editor-renderer` on gcr.io

File: minikube/athene2/main.tf [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+  image_pull_policy          = "Never"
+}
+
+#####################################################################
+# athene2
+#####################################################################
+
+module "legacy-editor-renderer" {
+  source            = "./../../modules/legacy-editor-renderer"
+  image             = "serlo/athene2-legacy-editor-renderer:latest"
```
Comment (inyono) [inline minikube/athene2/main.tf]: `serlo/legacy-editor-renderer:latest`

File: modules/athene2/definitions.php.tpl [hunk 6]
```diff
@@ -24,8 +24,8 @@
 $env = 'production';
 
 $assets = [
-    'legacy_editor_renderer' => 'http://' . getenv("LEGACY_EDITOR_RENDERER_SERVICE_SERVICE_HOST") . ':' . getenv("LEGACY_EDITOR_RENDERER_SERVICE_SERVICE_PORT") . '/',
-    'editor_renderer' => 'http://' . getenv("EDITOR_RENDERER_SERVICE_SERVICE_HOST") . ':' . getenv("EDITOR_RENDERER_SERVICE_SERVICE_PORT") . '/',
+    'legacy_editor_renderer' => 'http://legacy-editor-renderer-service.athene2:3000/',
```
Comment (inyono) [inline modules/athene2/definitions.php.tpl]: Why did that change? Does this still work for google cloud `dev` / `staging` / `production`?

File: container/dbdump/Dockerfile (hunk 1 above)
Comment (rbock44) [reply container/dbdump/Dockerfile]: kronjobs are in beta and are not supported by terraform at the moment.

File: live/dev/main.tf (hunk 2 above)
Comment (rbock44) [reply live/dev/main.tf]: it is working so I would leave I had problems not setting the host.

File: minikube/athene2/main.tf (hunk 3 above)
Comment (rbock44) [reply minikube/athene2/main.tf]: I removed both of it now.

File: minikube/athene2/main.tf (hunk 4 above)
Comment (rbock44) [reply minikube/athene2/main.tf]: OK will change the local image names to be same.

Comment (rbock44) [reply minikube/athene2/main.tf]: Hm. Nach etwas nachdenken. Vielleicht wäre es schon sinnvoll quasi einen Namespace im Namen zu haben. Es gibt ja Projekte wie z.B. KPI die ebenfalls container images bauen. Da könnte es bald mal zu kollisionen kommen. Was meinst Du? Falls Du keine Bedenken hast ändere ich das.

File: modules/athene2/definitions.php.tpl (hunk 6 above)
Comment (rbock44) [reply modules/athene2/definitions.php.tpl]: Wie wir annehmen ist die ENV variable ein GCE Feature. Im Minikube hat das nicht funktioniert. Die  obige schreibweise sollte aber auch in GCE funktionieren.

File: minikube/athene2/main.tf (hunk 5 above)
Comment (rbock44) [reply minikube/athene2/main.tf]: changed to serlo/legacy-editor-renderer

File: minikube/athene2/main.tf (hunk 4 above)
Comment (rbock44) [reply minikube/athene2/main.tf]: changes names as request.

**Pull request merged**